tests/DfE.FindInformationAcademiesTrusts.UnitTests/Pages/Trusts/Contacts/EditSfsoLeadModelTests.cs

```diff
@@ -53,4 +55,63 @@ public async Task OnGetAsync_loads_the_correct_name_and_email()
         _sut.Name.Should().Be(_sfsoLead.FullName);
         _sut.Email.Should().Be(_sfsoLead.Email);
     }
+
+    [Theory]
+    [InlineData(true, true)]
+    [InlineData(true, false)]
+    [InlineData(false, true)]
+    [InlineData(false, false)]
+    public async Task OnPostAsync_sets_ContactUpdated_to_true_when_validation_is_correct(bool nameUpdated,
+        bool emailUpdated)
+    {
+        _sut.TrustSummary = _fakeTrust;
+        _mockTrustService
+            .Setup(r => r.UpdateContactAsync(1234, It.IsAny<string>(), It.IsAny<string>(), ContactRole.SfsoLead))
+            .ReturnsAsync(new TrustContactUpdatedServiceModel(emailUpdated, nameUpdated));
+
+        var result = await _sut.OnPostAsync();
+
+        result.Should().BeOfType<RedirectToPageResult>();
+        
+        if (nameUpdated || emailUpdated)
+        {
+            _sut.ContactUpdatedMessage.Should()
+                .Contain("SFSO (Schools financial support and oversight) lead");
+        }
+
+        if (nameUpdated && emailUpdated)
```

Comment:
`if`s around assertions is an antipattern in testing. We could instead move the expected message up to the `InlineData` which would also make the test easier to read